newm-chain-db/src/main/kotlin/io/newm/chain/database/repository/ChainRepository.kt

```diff
@@ -0,0 +1,260 @@
+package io.newm.chain.database.repository
+
+import io.newm.chain.config.Config
+import io.newm.chain.database.entity.ChainBlock
+import io.newm.chain.database.table.ChainTable
+import io.newm.chain.database.table.KeysTable
+import io.newm.chain.database.table.PaymentStakeAddressTable
+import io.newm.chain.database.table.TransactionDestAddressTable
+import io.newm.chain.util.Blake2b
+import io.newm.chain.util.hexToByteArray
+import io.newm.chain.util.toHexString
+import io.newm.kogmios.protocols.model.PointDetail
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.greaterEq
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+object ChainRepository {
```

Comment:
What do you think of making all of these Repositories classes instead of objects and use dependency injection?

Reply:
What would be the advantage of DI when they are singletons?  I'm either going to use the Koin singleton to get an instance of the repository, or just use the Repository as a singleton itself.

The advantage of doing DI would be if we're planning to inject the repository as an interface and then our specific implementation here would be Postgresql. 

I guess it makes sense for future proofing. Let's throw a ticket in the backlog for it, but I don't want to slow down the MVP right now.

Reply:
My reasoning was more about consistency, since we use Koin to inject all other resposotories.

Reply:
newm-chain doesn't use Koin at all yet.